### PR TITLE
[Ncl-2139] added new options to cli tool

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

**Comment by thescouser89** [comment]
retest this please

**Comment by janinko** [comment]
Please update the readme to cover both new options.

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 1]
```diff
@@ -336,6 +458,7 @@ scm_report_align() {
         esac
         shift
     done
+    printf '%s\n' "${products[@]}"
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: Forgotten debug message?

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 2]
```diff
@@ -128,22 +132,61 @@ check() {
...
+                productNames="$productNames$2"
+                shift
+                ;;
+            --productIDs)
+                [ -n "$productVersionIds" ] && productVersionIds="$productVersionIds,"
+                productVersionIds="$productVersionIds$2"
+                shift
+                ;;
+             *) gav="$1"
+                shift ;;
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: Are you sure about this shift? I think the GAV is only one argument

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 3]
```diff
@@ -128,22 +132,61 @@ check() {
...
+        esac
         shift
+    done
+    echo $gav
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: Forgotten debug message?

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 4]
```diff
@@ -153,25 +196,68 @@ parseGAV() {
...
+            --productIDs)
+                [ -n "$productVersionIds" ] && productVersionIds="$productVersionIds,"
+                productVersionIds="$productVersionIds$2"                    
+                shift
+                ;;
+             *) parsedGAV=`parseGAV $1`
+                matchGAV $parsedGAV
+                gavs="$gavs `formatGAVjson`" 
+                input=false;
+                shift ;;
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: Are you sure about this shift? I think the GAV is only one argument

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 5]
```diff
@@ -153,25 +196,68 @@ parseGAV() {
...
+    msize=$(expr length $productVersionIds)
+    if [ $msize -eq 2 ]; then
+        query="$query \"productVersionIds\": []"
+    else
+        query="$query \"productVersionIds\": [$productVersionIds]"
+    fi
+    query="$query, \"gavs\": [$gavs]"
+    query="$query }"
+
+    echo $query
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: Forgotten debug message?

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 6]
```diff
@@ -153,25 +196,68 @@ parseGAV() {
...
+
     tmpfile=`gettmpfile`
-    post "reports/lookup/gavs" "$query" > $tmpfile
-    cat $tmpfile | prettyPrint lookup
+    post "reports/lookup/gavs" "$query" >> $tmpfile
+
+    cat $tmpfile #| prettyPrint lookup
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: remove the # again

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 7]
```diff
@@ -257,24 +343,40 @@ pom_bw_junit_xml() {
...
-        type="json"
-        shift
-    fi
-
+    local productNames=""
+    local productVersionIds=""
     local scm="$1"
     local tag="$2"
     local pom_path="$3"
+
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: what about  ``shift 3`` after this, so you don't go over $1, $2 and $3 in the while loop?

File: cli-wrap/src/main/resources/cli/listings.sh [hunk 8]
```diff
@@ -285,21 +387,41 @@ scm_report() {
...
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.sh]: what about  ``shift 3`` after this, so you don't go over $1, $2 and $3 in the while loop?

**Comment by stanlyDoge** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

**Comment by janinko** [comment]
When you update documentation => :+1:

**Pull request merged**